libraries/core_libs/node/src/node.cpp

```diff
@@ -185,6 +185,30 @@ void FullNode::start() {
           }
         },
         *rpc_thread_pool_);
+
+    // Subscription to process hardforks
+    final_chain_->block_applying.subscribe([&](uint64_t block_num) {
```

Comment:
What is the advantage to use emitter and trigger the hf function here vs not using any emitter and trigger hf function in `finalize` function ?

Reply:
We need to apply that before block is finalized and state commited, so I created one another emitter. 
Also it is in the node to be more generic thing. Because we can not change a lot of things from final_chain
Also I just realised that I forgot about config updation. Config file should be updated from that part of code also, not just inmemory config. So it is another one argument for it to be here I guess